core/src/main/java/com/linecorp/armeria/server/ServiceConfigBuilder.java

```diff
@@ -348,7 +358,8 @@ ServiceConfig build(ServiceNaming defaultServiceNaming,
                 multipartUploadsLocation != null ? multipartUploadsLocation : defaultMultipartUploadsLocation,
                 ImmutableList.copyOf(shutdownSupports),
                 mergeDefaultHeaders(virtualHostDefaultHeaders.toBuilder(), defaultHeaders.build()),
-                requestIdGenerator != null ? requestIdGenerator : defaultRequestIdGenerator, errorHandler);
+                requestIdGenerator != null ? requestIdGenerator : defaultRequestIdGenerator, errorHandler,
+                contextHook != null ? contextHook : () -> (AutoCloseable) () -> {});
```

Comment:
Could we create singleton `NOOP_CONTEXT_HOOK` so that `RequestContext.hook(Supplier)` ignore that?
```java
class {
  public void hook(Supplier<? extends AutoCloseable> contextHook) {
    if (contextHook == NOOP_CONTEXT_HOOK) {
      return;
    }
    ...
  }
}
```

Reply:
> Could we create singleton NOOP_CONTEXT_HOOK so that RequestContext.hook(Supplier) ignore that?

Yes, we could. I'll do that.

There are many locations which `() -> () -> {}` is implemented.
Should we replace these all implementations to `NOOP_CONTEXT_HOOK`?
If so, is it good to define `NOOP_CONTEXT_HOOK` in several files?